epoxy-annotations/src/main/java/com/airbnb/epoxy/EpoxyAttribute.java

```diff
@@ -7,25 +7,77 @@
 import java.lang.annotation.Target;
 
 /**
- * Used to annotate fields on EpoxyModel classes in order to generate a
- * subclass of that model with getters, setters, equals, and hashcode for the annotated fields.
+ * Used to annotate fields on EpoxyModel classes in order to generate a subclass of that model with
+ * getters, setters, equals, and hashcode for the annotated fields.
  */
 @Target(ElementType.FIELD)
 @Retention(RetentionPolicy.CLASS)
 public @interface EpoxyAttribute {
   /**
+   * Options that can be included on the attribute to affect how the model's generated class is
+   * created.
+   */
+  enum Option {
+    /**
+     * A getter is generated for this attribute by default. Add this option to prevent a getter from
+     * being generated.
+     */
+    NoGetter,
+    /**
+     * A setter is generated for this attribute by default. Add this option to prevent a setter from
+     * being generated.
+     */
+    NoSetter,
+    /**
+     * By default every attribute's hashCode method is called when determining the model's state.
+     * This option can be used to exclude an attribute's hashCode from contributing to the state.
+     * <p>
+     * This is useful for objects that may change without actually changing the model's state. A
+     * common case is an anonymous click listener that gets recreated with every bind call.
+     * <p>
+     * When this is used, the attribute will affect the model state solely based on whether it is
+     * null or non null.
+     */
+    DoNotHash,
+    /**
+     * This is meant to be used in conjunction with {@link PackageEpoxyConfig#requireHashCode()}.
+     * When that is enabled every attribute must implement hashCode. However, there are some valid
+     * cases where the attribute type does not implement hashCode, but it should still be hashed at
+     * runtime and contribute to the model's state. Use this option on an attribute in that case to
+     * tell the processor to let it pass the hashCode validation.
+     * <p>
+     * An example case is AutoValue classes, where the generated class correctly implements hashCode
+     * at runtime.
+     * <p>
+     * If you use this it is your responsibility to ensure that the object assigned to the attribute
+     * at runtime correctly implements hashCode. If you don't want the attribute to contribute to
+     * model state you should use {@link Option#DoNotHash} instead.
+     */
+    AllowMissingHash
```

Comment:
`requireHashCode` doesn't make as much sense on the attribute, since a hashCode is always required (unless DoNotHash is used). Maybe a better name for the attribute option could be `IgnoreRequireHashCode` so that the name ties back better to the config option. I don't mind it being verbose since it should rarely be used, and I think having a name that is clear is important here. What do you think of `IgnoreRequireHashCode` vs `AllowMissingHash` (or `AllowMissingHashCode`)? I want the name to say "I know this field type doesn't implement hashCode, but it's ok and shouldn't fail the hashCode requirement check because I know the object used at runTime _will_ properly implement hashCode".